src/connector/src/parser/mod.rs

```diff
@@ -328,13 +330,18 @@ impl SourceStreamChunkRowWriter<'_> {
                         // TODO: decide whether the error should not be ignored (e.g., even not a valid Debezium message)
                         // TODO: not using tracing span to provide `split_id` and `offset` due to performance concern,
                         //       see #13105
-                        tracing::warn!(
-                            %error,
-                            split_id = self.row_meta.as_ref().map(|m| m.split_id),
-                            offset = self.row_meta.as_ref().map(|m| m.offset),
-                            column = desc.name,
-                            "failed to parse non-pk column, padding with `NULL`"
-                        );
+                        static LOG_SUPPERSSER: LazyLock<LogSuppresser> =
+                            LazyLock::new(|| LogSuppresser::default());
+                        if let Ok(suppressed_count) = LOG_SUPPERSSER.check() {
```

Comment:
What about making `check` return `Option<NonZeroUsize>`? In this way the `suppressed_count` won't be there if it's zero.

https://docs.rs/tracing-core/0.1.32/src/tracing_core/field.rs.html#749

Reply:
In this way, I suppose the caller needs to do pattern match to differentiate 3 cases? 

```
match result {
   case Ok(Some(NonZero(suppressed_count)) => print a message with suppressed count
   case Ok(None) => print a message without suppressed count
   case Err(_) => suppress
}
```

Too verbose for me

Reply:
I guess no need. `Option<NonZeroUsize>` can be directly used as a field for tracing events.